DoorGate/DoorPresenter.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  DoorPresenter.swift
+//  
+//
+//  Created by Yuliani Noriega on 1/26/18.
+//
+
+struct DoorPresenter {
+    let doorEntity = DoorEntity()
+    
+    mutating func buttonTap(command:DoorCommand) {
+        let state = doorEntity.state.value //easier to read in the if statements below
+        
+        if command == .goingIn && state < DoorState.locked.rawValue {
```

Comment:
Are we always evaluating against .rawValue? Is there a way to not have to do this syntactically?

Reply:
No, doing DoorState.locked is a enum and when comparing against state it's expecting an Int
<img width="595" alt="screen shot 2018-02-05 at 10 11 41 am" src="https://user-images.githubusercontent.com/938211/35811705-20a5b81c-0a5d-11e8-9a21-031029336f66.png">